public/app/features/explore/ContentOutline/ContentOutline.tsx

```diff
@@ -0,0 +1,82 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { useToggle } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { useStyles2, ToolbarButton } from '@grafana/ui';
+
+import { useContentOutlineContext } from './ContentOutlineContext';
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    wrapper: css({
+      label: 'wrapper',
+      position: 'relative',
+      display: 'flex',
+      justifyContent: 'center',
+      height: '100%',
+      backgroundColor: theme.colors.background.primary,
+      marginRight: theme.spacing(1),
+      zIndex: theme.zIndex.sidemenu,
+    }),
+    content: css({
+      label: 'content',
+      position: 'sticky',
+      top: 0,
+      height: '85vh',
+    }),
+    buttonStyles: css({
+      textAlign: 'left',
+      width: '100%',
+    }),
+  };
+};
+
+const ContentOutline = () => {
+  const [expanded, toggleExpanded] = useToggle(false);
+  const styles = useStyles2((theme) => getStyles(theme));
+  const { outlineItems } = useContentOutlineContext();
+
+  const scrollIntoView = (ref: HTMLElement | null, buttonTitle: string) => {
+    ref?.scrollIntoView({ behavior: 'smooth' });
+    reportInteraction('grafana_explore_content_outline_item_clicked', {
+      scrollButtonClicked: buttonTitle,
+    });
+  };
+
+  const toggle = () => {
+    toggleExpanded();
+    reportInteraction('grafana_explore_content_outline_toggle_expand', {
+      expanded: !expanded,
+    });
+  };
+
+  return (
+    <div className={styles.wrapper}>
+      <div className={styles.content}>
+        <ToolbarButton
+          className={styles.buttonStyles}
+          icon={expanded ? 'angle-left' : 'angle-right'}
+          onClick={toggle}
+          tooltip={!expanded ? 'Show Content Outline' : undefined}
+        >
+          {expanded && 'Hide Content Outline'}
```

Comment:
we generally use sentence case in the UI
```suggestion
          tooltip={!expanded ? 'Show content outline' : undefined}
        >
          {expanded && 'Hide content outline'}
```